<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>programClass</class>
 <widget class="QMainWindow" name="programClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>影像處理</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="lbl_image">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>10</y>
      <width>450</width>
      <height>400</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="Histogram_btn">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>160</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Histogram</string>
    </property>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>21</y>
      <width>133</width>
      <height>125</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QSplitter" name="x_splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QLabel" name="x_label">
        <property name="text">
         <string>x</string>
        </property>
       </widget>
       <widget class="QSlider" name="x_slider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSpinBox" name="x_spinbox"/>
      </widget>
     </item>
     <item>
      <widget class="QSplitter" name="y_splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QLabel" name="y_label">
        <property name="text">
         <string>y</string>
        </property>
       </widget>
       <widget class="QSlider" name="y_slider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSpinBox" name="y_spinbox"/>
      </widget>
     </item>
     <item>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QLabel" name="w_label">
        <property name="text">
         <string>w</string>
        </property>
       </widget>
       <widget class="QSlider" name="w_slider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSpinBox" name="w_spinbox"/>
      </widget>
     </item>
     <item>
      <widget class="QSplitter" name="h_splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QLabel" name="h_label">
        <property name="text">
         <string>h</string>
        </property>
       </widget>
       <widget class="QSlider" name="h_slider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QSpinBox" name="h_spinbox"/>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_ROI">
       <property name="text">
        <string>ROI</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QSplitter" name="splitter_3">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>220</y>
      <width>116</width>
      <height>31</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
    <widget class="QLabel" name="BV_label">
     <property name="acceptDrops">
      <bool>true</bool>
     </property>
     <property name="statusTip">
      <string/>
     </property>
     <property name="text">
      <string>Binary_Value</string>
     </property>
    </widget>
    <widget class="QSplitter" name="splitter_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QSlider" name="BV_slider">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="BV_spinbox"/>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="File">
    <property name="title">
     <string> 檔案</string>
    </property>
    <addaction name="Open"/>
    <addaction name="Save"/>
   </widget>
   <addaction name="File"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionopen"/>
   <addaction name="separator"/>
   <addaction name="actionsave"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="Open">
   <property name="text">
    <string>Open </string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="Save">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionopen">
   <property name="icon">
    <iconset>
     <normaloff>../../image/open.png</normaloff>../../image/open.png</iconset>
   </property>
   <property name="text">
    <string>open</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionsave">
   <property name="icon">
    <iconset>
     <normaloff>../../image/save.png</normaloff>../../image/save.png</iconset>
   </property>
   <property name="text">
    <string>save</string>
   </property>
   <property name="toolTip">
    <string>Sace</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="program.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>Open</sender>
   <signal>triggered()</signal>
   <receiver>programClass</receiver>
   <slot>OpenImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionopen</sender>
   <signal>triggered()</signal>
   <receiver>programClass</receiver>
   <slot>OpenImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>348</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_ROI</sender>
   <signal>clicked()</signal>
   <receiver>programClass</receiver>
   <slot>ROI()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>577</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>348</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>x_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>531</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>x_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>y_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>y_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>w_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>w_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>w_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>w_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>h_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>h_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>580</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>638</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>h_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>h_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>155</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Histogram_btn</sender>
   <signal>clicked()</signal>
   <receiver>programClass</receiver>
   <slot>Histogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>607</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BV_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>programClass</receiver>
   <slot>binary_value()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>669</x>
     <y>393</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BV_slider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>BV_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>610</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>670</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>BV_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BV_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>670</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>610</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenImage()</slot>
  <slot>ROI()</slot>
  <slot>Histogram()</slot>
  <slot>binary_value()</slot>
 </slots>
</ui>
